deepmd/utils/data.py

```diff
@@ -129,6 +129,16 @@ def __init__(
         self.shuffle_test = shuffle_test
         # set modifier
         self.modifier = modifier
+        # calculate prefix sum for get_item method
+        self.nframes = 0
+        i = 1
+        # The prefix sum stores the range of indices contained in each directory, which is needed by get_item method
+        self.prefix_sum = [0] * (len(self.dirs) + 1)
+        for item in self.dirs:
+            frames = self._get_nframes(item)
+            self.prefix_sum[i] = self.prefix_sum[i - 1] + frames
+            i += 1
+            self.nframes += frames
```

Comment:
It looks you just rewrote np.cumsum and np.sum

Reply:
```suggestion
        # calculate prefix sum for get_item method
        frames_list = [self._get_nframes(item) for item in self.dirs]
        self.nframes = np.sum(frames_list)
        # The prefix sum stores the range of indices contained in each directory, which is needed by get_item method
        self.nframes = np.cumsum([0] + frames_list).tolist()
```

@CaRoLZhangxy Plz try if this is OK

Reply:
Or use `itertools.accumulate` and `sum` without numpy